### PR TITLE
Dataset db model

### PR DESCRIPTION
**Pull request by KapRom**
Added Database model for metadata and tests.

### CODE REVIEW

**Comment by KapRom** [comment]
[Create database model for data files links storage](https://trello.com/c/lOfWweNU/171-create-database-model-for-data-files-links-storage)

File: .gitattributes [hunk 1]
```diff
@@ -1,63 +1,77 @@
-###############################################################################
-# Set default behavior to automatically normalize line endings.
-###############################################################################
-* text=auto
+using System;
```
Comment (Daroslaw) [inline .gitattributes]: I don't think that is supposed to be here

File: src/Spectre.Database.Tests/QueryTests.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            
+            var mockContext = new Mock<Context>();
+            mockContext.Setup(c => c.Datasets).Returns(mockSet.Object);
+
+
+            var service = new PathFinder(mockContext.Object);
+            var dataset1 = service.ReturnForHash("1");
+            var dataset2 = service.ReturnForFriendlyName("Bajer2");
+
+            NUnit.Framework.Assert.AreEqual("path1", dataset1);
```
Comment (Daroslaw) [inline src/Spectre.Database.Tests/QueryTests.cs]: Having `using NUnit.Framework` do you need to do the entire namespace expansion to get to desired method?

File: src/Spectre.Database/Utils/PathFinder.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+{
+    /// <summary>
+    /// Class for finding the path of chosen property.
+    /// </summary>
+    public class PathFinder
+    {
+        /// <summary>
+        /// Variable for returning location from hash
+        /// </summary>
+        private static string _locationfromhash;
```
Comment (Daroslaw) [inline src/Spectre.Database/Utils/PathFinder.cs]: Is it really needed to store in privates what can be directly returned to the user? They're privates after all, so user cannot check the current state/last returned paths.
Unless there is really a neccessity that I am not aware of.

File: src/Spectre.Database/Utils/PathFinder.cs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+        /// <summary>
+        /// Variable for returning location from friendly name
+        /// </summary>
+        private static string _locationfromfriendlyname;
```
Comment (Daroslaw) [inline src/Spectre.Database/Utils/PathFinder.cs]: As above.

File: src/Spectre.Database/Utils/PathFinder.cs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+        /// <summary>
+        /// The location
+        /// </summary>
+        private Dataset _path = new Dataset();
```
Comment (Daroslaw) [inline src/Spectre.Database/Utils/PathFinder.cs]: As above. Also misleading variable name, as Dataset entity does not directly describe the path of given dataset.

File: src/Spectre.Database/Utils/PathFinder.cs [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        /// <returns>
+        /// Returns location having hash.
+        /// </returns>
+        public string ReturnForHash(string hash)
+        {
+            _path = _context.Datasets
+                    .Where(b => b.Hash == hash)
+                    .FirstOrDefault();
+
+            PathFinder._locationfromhash = _path.Location.ToString();
```
Comment (Daroslaw) [inline src/Spectre.Database/Utils/PathFinder.cs]: My proposition is to keep `_path` local to this method and return directly `_path.Location.ToString()`.

File: src/Spectre.Database/App.config [hunk 7]
```diff
@@ -0,0 +1,17 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (FreexD) [inline src/Spectre.Database/App.config]: This config file is repeated. We could consider extracting it's functionality into a service class (under an interface) that will provide connections for desired DBs.

File: src/Spectre.Database/Utils/PathFinder.cs [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using Spectre.Database.Entities;
+
+namespace Spectre.Database.Utils
+{
+    /// <summary>
+    /// Class for finding the path of chosen property.
+    /// </summary>
+    public class PathFinder
```
Comment (FreexD) [inline src/Spectre.Database/Utils/PathFinder.cs]: PathFinder is some sort of Repository class. It accesses the DB to return Dataset location by name or hash. Why not extract it to an interface and make it a service in case other classes would need to use that as well.

File: src/Spectre.Database/Utils/PathFinder.cs [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+        /// <summary>
+        /// The context description
+        /// </summary>
+        private Context _context;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PathFinder"/> class.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        public PathFinder(Context context)
```
Comment (FreexD) [inline src/Spectre.Database/Utils/PathFinder.cs]: We would need to inject it into all Repository classes. We should think of a way to make it without repeating that constructor everywhere.

File: src/Spectre.Database.Tests/QueryTests.cs [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+            mockSet.As<IQueryable<Dataset>>().Setup(m => m.Expression).Returns(data.Expression);
+            mockSet.As<IQueryable<Dataset>>().Setup(m => m.ElementType).Returns(data.ElementType);
+            mockSet.As<IQueryable<Dataset>>().Setup(m => m.GetEnumerator()).Returns(data.GetEnumerator());
+            
+            var mockContext = new Mock<Context>();
+            mockContext.Setup(c => c.Datasets).Returns(mockSet.Object);
+
+
+            var service = new PathFinder(mockContext.Object);
+            var dataset1 = service.ReturnForHash("1");
```
Comment (FreexD) [inline src/Spectre.Database.Tests/QueryTests.cs]: It's dataset location not dataset.

File: src/Spectre.Database.Tests/QueryTests.cs [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+            mockSet.As<IQueryable<Dataset>>().Setup(m => m.Provider).Returns(data.Provider);
+            mockSet.As<IQueryable<Dataset>>().Setup(m => m.Expression).Returns(data.Expression);
+            mockSet.As<IQueryable<Dataset>>().Setup(m => m.ElementType).Returns(data.ElementType);
+            mockSet.As<IQueryable<Dataset>>().Setup(m => m.GetEnumerator()).Returns(data.GetEnumerator());
+            
+            var mockContext = new Mock<Context>();
+            mockContext.Setup(c => c.Datasets).Returns(mockSet.Object);
+
+
+            var service = new PathFinder(mockContext.Object);
```
Comment (FreexD) [inline src/Spectre.Database.Tests/QueryTests.cs]: Asks for an interface and an instance provider to replace new.

File: src/Spectre.Database/Utils/Context.cs [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System.Data.Entity;
+using Spectre.Database.Entities;
+
+namespace Spectre.Database.Utils
```
Comment (FreexD) [inline src/Spectre.Database/Utils/Context.cs]: DbContexts are not utils IMHO. They deserve a package name.

File: src/Spectre.Database/Utils/PathFinder.cs [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Spectre.Database.Entities;
+
+namespace Spectre.Database.Utils
```
Comment (FreexD) [inline src/Spectre.Database/Utils/PathFinder.cs]: I'd call that Repository.

File: src/Spectre.Database.Tests/QueryTests.cs (hunk 10 above)
Comment (FreexD) [inline src/Spectre.Database.Tests/QueryTests.cs]: Each method 'ReturnForHash' and 'ReturnForFriendlyName' should be tested separately as separate services. You can to that using the [SetUp] annotation. You are also missing corner case tests.

File: src/Spectre.Database/Utils/PathFinder.cs (hunk 3 above)
Comment (FreexD) [inline src/Spectre.Database/Utils/PathFinder.cs]: +1 Why even keep those privates here?

File: src/Spectre.Database/Utils/PathFinder.cs [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+        {
+            _context = context;
+        }
+
+        /// <summary>
+        /// Returns for hash.
+        /// </summary>
+        /// <param name="hash">The hash.</param>
+        /// <returns>
+        /// Returns location having hash.
```
Comment (FreexD) [inline src/Spectre.Database/Utils/PathFinder.cs]: What if no dataset of given hash is in the DB.

File: src/Spectre.Database/Utils/PathFinder.cs [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+
+            return PathFinder._locationfromhash;
+        }
+
+        /// <summary>
+        /// Returns the name of friendly name.
+        /// </summary>
+        /// <param name="friendlyname">The friendlyname.</param>
+        /// <returns>
+        /// Returns location having friendly name.
```
Comment (FreexD) [inline src/Spectre.Database/Utils/PathFinder.cs]: What if no dataset of given friendly name is in the DB.

File: src/Spectre.Database.Tests/QueryTests.cs (hunk 2 above)
Comment (KapRom) [reply src/Spectre.Database.Tests/QueryTests.cs]: When using only Assert, says that it's ambiguous. NUnit.Framework is needed to perform tests.

File: src/Spectre.Database.Tests/QueryTests.cs [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+            
+            var mockContext = new Mock<Context>();
+            mockContext.Setup(c => c.Datasets).Returns(mockSet.Object);
+            
+            var service = new PathFinder(mockContext.Object);
+            var dataset1 = service.ReturnForHash("1");
+            var dataset2 = service.ReturnForFriendlyName("Bajer2");
+            var dataset3 = service.ReturnForHash("4");
+
+            NUnit.Framework.Assert.AreEqual("path1", dataset1);
```
Comment (Regis128) [inline src/Spectre.Database.Tests/QueryTests.cs]: use assertMultiple or better to create 3 separate tests

File: src/Spectre.Service/Spectre.Service_BACKUP_11200.csproj [hunk 17]
```diff
@@ -0,0 +1,109 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (Regis128) [inline src/Spectre.Service/Spectre.Service_BACKUP_11200.csproj]: Why there are so many of BACKUP files?

File: src/Spectre.Service/Spectre.Service_BACKUP_8796.csproj [hunk 18]
```diff
@@ -0,0 +1,109 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (Regis128) [inline src/Spectre.Service/Spectre.Service_BACKUP_8796.csproj]: As above

File: src/Spectre.Service/Spectre.Service_REMOTE_11200.csproj [hunk 19]
```diff
@@ -0,0 +1,100 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (Regis128) [inline src/Spectre.Service/Spectre.Service_REMOTE_11200.csproj]: Again with remote files

File: src/Spectre.sln [hunk 20]
```diff
@@ -103,112 +103,164 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Spectre.Dependencies", "Spe
 EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Any CPU = Debug|Any CPU
```
Comment (Daroslaw) [inline src/Spectre.sln]: There should be no platform configuration for `Any CPU`, only for `x64`. This relates to everything that is below in this file.

File: src/Spectre.Service/Spectre.Service.csproj [hunk 21]
```diff
@@ -56,11 +56,13 @@
   <ItemGroup>
```
Comment (Daroslaw) [inline src/Spectre.Service/Spectre.Service.csproj]: Please get rid of any unneccessary git-related copies of this file.

File: src/Spectre.Service/PathFinderService.cs [hunk 22]
```diff
@@ -0,0 +1,60 @@
...
+    using System.Threading.Tasks;
+    using Spectre.Database.Contexts;
+    using Spectre.Database.Utils;
+    using Spectre.Service.Abstract;
+
+    /// <summary>
+    /// PathFinder service.
+    /// </summary>
+    /// <seealso cref="Spectre.Service.Abstract.IPathFinderService" />
+    internal class PathFinderService : IPathFinderService
```
Comment (Daroslaw) [inline src/Spectre.Service/PathFinderService.cs]: Shouldn't this be a public class?

File: src/Spectre.Database/Contexts/Context.cs [hunk 23]
```diff
@@ -35,5 +35,21 @@ public class Context : DbContext
...
+        /// </value>
+        public virtual DbSet<Operation> OperationDatasets { get; set; }
+
+        /// <summary>
+        /// Gets or sets the owner datasets.
+        /// </summary>
+        /// <value>
+        /// The owner datasets.
+        /// </value>
+        public virtual DbSet<OwnerDataset> OwnerDatasets { get; set; }
```
Comment (Daroslaw) [inline src/Spectre.Database/Contexts/Context.cs]: Same as above

File: src/Spectre.Database/Entities/OwnerDataset.cs [hunk 24]
```diff
@@ -0,0 +1,94 @@
...
+    {
+        /// <summary>
+        /// The identifier
+        /// </summary>
+        /// <value>
+        /// The custom identifier.
+        /// </value>
+        [Key]
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public string DatasetId { get; set; }
```
Comment (Daroslaw) [inline src/Spectre.Database/Entities/OwnerDataset.cs]: Also misleading name.

File: src/Spectre.Database/Entities/OwnerDataset.cs [hunk 25]
```diff
@@ -0,0 +1,94 @@
...
+        /// </value>
+        public string Checksum { get; set; }
+
+        /// <summary>
+        /// Gets or sets the owner.
+        /// </summary>
+        /// <value>
+        /// The owner.
+        /// </value>
+        public string Owner { get; set; }
```
Comment (Daroslaw) [inline src/Spectre.Database/Entities/OwnerDataset.cs]: Duplicated with Owner in Dataset entity.

File: src/Spectre.sln [hunk 26]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
-VisualStudioVersion = 15.0.26430.16
+VisualStudioVersion = 15.0.26730.3
```
Comment (Daroslaw) [inline src/Spectre.sln]: Unneccessary change

File: src/Spectre.Service/PathFinderService.cs (hunk 22 above)
Comment (KapRom) [reply src/Spectre.Service/PathFinderService.cs]: no

File: src/Spectre.Database/Utils/Finder.cs [hunk 27]
```diff
@@ -0,0 +1,101 @@
...
+    {
+        /// <summary>
+        /// The context description
+        /// </summary>
+        private Context _context;
+
+        /// <summary>
+        /// The dataset
+        /// </summary>
+        private Dataset _dataset = new Dataset();
```
Comment (gmrukwa) [inline src/Spectre.Database/Utils/Finder.cs]: Why are we creating new instance here?

File: src/Spectre.Database/Utils/Finder.cs [hunk 28]
```diff
@@ -0,0 +1,101 @@
...
+        /// </returns>
+        public virtual string ReturnForHash(string hash)
+        {
+            if (_context.Datasets.Any(o => o.Hash == hash))
+            {
+                _dataset = _context.Datasets
+                    .Where(b => b.Hash == hash)
+                    .FirstOrDefault();
+
+                return _dataset.UploadNumber.ToString();
```
Comment (gmrukwa) [inline src/Spectre.Database/Utils/Finder.cs]: `UploadNumber` is already a `string`, am I wrong? What if no dataset of a given hash is found?

File: src/Spectre.Database/Utils/Finder.cs [hunk 29]
```diff
@@ -0,0 +1,101 @@
...
+        /// </returns>
+        public virtual string ReturnForFriendlyName(string friendlyname)
+        {
+            if (_context.Datasets.Any(o => o.FriendlyName == friendlyname))
+            {
+                _dataset = _context.Datasets
+                    .Where(b => b.FriendlyName == friendlyname)
+                    .FirstOrDefault();
+
+                return _dataset.UploadNumber.ToString();
```
Comment (gmrukwa) [inline src/Spectre.Database/Utils/Finder.cs]: As above.

File: src/Spectre.Service/Abstract/IFinderService.cs [hunk 30]
```diff
@@ -0,0 +1,53 @@
...
+    public interface IFinderService
+    {
+        /// <summary>
+        /// Finds the hash using friendly name.
+        /// </summary>
+        /// <param name="hash">The hash.</param>
+        /// <returns>
+        /// Friendly name basing on hash
+        /// </returns>
+        string FindHash(string hash);
```
Comment (gmrukwa) [inline src/Spectre.Service/Abstract/IFinderService.cs]: Consider renames as suggested in the `Finder`.

File: src/Spectre.Service/FinderService.cs [hunk 31]
```diff
@@ -0,0 +1,60 @@
...
+    using System.Threading.Tasks;
+    using Spectre.Database.Contexts;
+    using Spectre.Database.Utils;
+    using Spectre.Service.Abstract;
+
+    /// <summary>
+    /// PathFinder service.
+    /// </summary>
+    /// <seealso cref="Spectre.Service.Abstract.IFinderService" />
+    internal class FinderService : IFinderService
```
Comment (gmrukwa) [inline src/Spectre.Service/FinderService.cs]: Consider class name change, as suggested previously.

File: src/Spectre.Service/Spectre.Service.csproj (hunk 21 above)
Comment (gmrukwa) [reply src/Spectre.Service/Spectre.Service.csproj]: @Daroslaw, could you be more specific?

File: src/Spectre.Database.Tests/packages.config [hunk 32]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Castle.Core" version="4.1.1" targetFramework="net452" />
+  <package id="EntityFramework" version="6.1.3" targetFramework="net452" />
+  <package id="Moq" version="4.7.99" targetFramework="net452" />
+  <package id="MSTest.TestAdapter" version="1.1.11" targetFramework="net452" />
```
Comment (gmrukwa) [inline src/Spectre.Database.Tests/packages.config]: This should be removed, the same as the line below.

File: src/Spectre.Database.Tests/packages.config [hunk 33]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Castle.Core" version="4.1.1" targetFramework="net452" />
```
[the remark this answers is not in the document]
Comment (KapRom) [reply src/Spectre.Database.Tests/packages.config]: Can't be uninstalled, mocking framework Moq depends on it.

[the remark this answers is not in the document]
Comment (gmrukwa) [reply src/Spectre.Database.Tests/packages.config]: That's what I wanted to know, thanks!

File: src/Spectre.Service/Spectre.Service.csproj (hunk 21 above)
Comment (Daroslaw) [reply src/Spectre.Service/Spectre.Service.csproj]: This is now resolved, those copies are gone

File: src/Spectre.Database.Tests/QueryTests.cs [hunk 34]
```diff
@@ -0,0 +1,116 @@
...
+            mockSet.As<IQueryable<Dataset>>().Setup(m => m.Provider).Returns(data.Provider);
+            mockSet.As<IQueryable<Dataset>>().Setup(m => m.Expression).Returns(data.Expression);
+            mockSet.As<IQueryable<Dataset>>().Setup(m => m.ElementType).Returns(data.ElementType);
+            mockSet.As<IQueryable<Dataset>>().Setup(m => m.GetEnumerator()).Returns(data.GetEnumerator());
+
+            _mockContext.Setup(c => c.Datasets).Returns(mockSet.Object);
+        }
+
+        [Test]
+        public void HashToUploadNumber_Test()
```
Comment (FreexD) [inline src/Spectre.Database.Tests/QueryTests.cs]: Unclear test naming. HashToUploadNumer_finds_proper_dataset_for_proper_data mabye? This comment is related to all unclear test names.

File: src/Spectre.Database.Tests/QueryTests.cs [hunk 35]
```diff
@@ -0,0 +1,116 @@
...
+        {
+            var service = new DatasetDetailsFinder(_mockContext.Object);
+
+            var dataset = service.FriendlyNameToUploadNumber("FriendlyName2");
+
+            Assert.AreEqual("UploadNumber2", dataset);
+        }
+
+        [Test]
+        public void Return_for_not_existing_hash_test()
```
Comment (FreexD) [inline src/Spectre.Database.Tests/QueryTests.cs]: Be consistent with the naming. HashToUploadNumer_returns_null_for_not_existing_hash_test mabye

File: src/Spectre.Database/App.config (hunk 7 above)
Comment (FreexD) [reply src/Spectre.Database/App.config]: It's fine by me as long as it is not repeated :)

File: src/Spectre.Database/Entities/Enumerations.cs [hunk 36]
```diff
@@ -0,0 +1,78 @@
...
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace Spectre.Database.Entities
```
Comment (FreexD) [inline src/Spectre.Database/Entities/Enumerations.cs]: Spectre.Database.Entities.Enums will make it easier for other team members to find enums declared here when they will need to develop something DB related ;)

File: src/Spectre.Database/Utils/DatasetDetailsFinder.cs [hunk 37]
```diff
@@ -0,0 +1,119 @@
...
+        {
+            _context = context;
+        }
+
+        /// <summary>
+        /// Query for translating hash to upload number.
+        /// </summary>
+        /// <param name="hash">The hash.</param>
+        /// <returns>
+        /// Returns UploadNumber for Hash.
```
Comment (FreexD) [inline src/Spectre.Database/Utils/DatasetDetailsFinder.cs]: Documentation should specify what is returned when hash is not found. It goes same for all the methods below.

File: src/Spectre.Database/App.config (hunk 7 above)
Comment (KapRom) [reply src/Spectre.Database/App.config]: config file is now deleted in database.tests project

File: src/Spectre.Database/Spectre.Database.csproj [hunk 38]
```diff
@@ -0,0 +1,70 @@
...
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{B5B60F79-653C-4E41-9721-06C478CE9E7C}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>Spectre.Database</RootNamespace>
+    <AssemblyName>Spectre.Database</AssemblyName>
+    <TargetFrameworkVersion>v4.5.2</TargetFrameworkVersion>
```
[the remark this answers is not in the document]
Comment (KapRom) [reply src/Spectre.Database/Spectre.Database.csproj]: merged with develop and no changes in here

File: src/Spectre.Database/Entities/Enums/Enumerations.cs [hunk 39]
```diff
@@ -0,0 +1,78 @@
...
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace Spectre.Database.Entities.Enums
+{
+    /// <summary>
+    /// Enumeration for types of the AccessType.
+    /// </summary>
+    public enum AccessType
```
Comment (gmrukwa) [inline src/Spectre.Database/Entities/Enums/Enumerations.cs]: I would still place each `enum` in separate file.

File: src/Spectre.Database/Utils/DatasetDetailsFinder.cs (hunk 37 above)
Comment (gmrukwa) [reply src/Spectre.Database/Utils/DatasetDetailsFinder.cs]: We should also go for .NET standard-conforming names in such a case, like `HashToUploadNumberOrDefault`, besides of entry in the docstrings, as stated in the conversation with @KapRom.

File: src/Spectre.Database/Utils/IDatasetDetailsFinder.cs [hunk 40]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        /// <summary>
+        /// Query for translating hash to upload number.
+        /// </summary>
+        /// <param name="hash">The hash.</param>
+        /// <returns>
+        /// Returns UploadNumber for Hash.
+        /// Null for not existing Hash.
+        /// </returns>
+        string HashToUploadNumber(string hash);
```
Comment (gmrukwa) [inline src/Spectre.Database/Utils/IDatasetDetailsFinder.cs]: Similar proposal for renames as above.

File: src/Spectre.Service/Abstract/IDatasetDetailsFinderService.cs [hunk 41]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        /// <summary>
+        /// Finds the hash using friendly name.
+        /// </summary>
+        /// <param name="hash">The hash.</param>
+        /// <returns>
+        /// Returns UploadNumber basing on hash.
+        /// Null for not existing hash.
+        /// </returns>
+        string HashToUploadNumber_Service(string hash);
```
Comment (gmrukwa) [inline src/Spectre.Service/Abstract/IDatasetDetailsFinderService.cs]: Why do we have `_Service` suffix here?

File: src/Spectre.Service/DatasetDetailsFinderService.cs [hunk 42]
```diff
@@ -0,0 +1,96 @@
...
+
+        /// <summary>
+        /// Service translating hash to upload number.
+        /// </summary>
+        /// <param name="hash">The hash.</param>
+        /// <returns>
+        /// Returns upload number basing on Hash.
+        /// Null for not existing hash.
+        /// </returns>
+        public string HashToUploadNumber_Service(string hash)
```
Comment (gmrukwa) [inline src/Spectre.Service/DatasetDetailsFinderService.cs]: As above.
